### ISSUE TITLE
Can't disable an env in a single file

### ISSUE THREAD

**Issue by callumlocke** [eslint/eslint#2614]
I've got a project all written in ES6, except for a single file that is in ES5.

In my .eslintrc:

```yaml
env:
  es6: true
  node: true
  browser: true
```

In the single ES5 file:

```js
/*eslint-env es6:false*/

const fn = () => {};
```

I expected to get an eslint error for the fat arrow, but there's no error.

**Comment by gyandeeps** [comment]
I just looked at the code and found out that there is no way you can disable environment but you can always enable environments. Basically they are not evaluated as boolean properties that why specifying `true` or `false` doesn't matter (this applies to in file declarations only).

**Comment by callumlocke** [comment]
So is this a bug? It seems like a common use case, e.g. writing a gulpfile in ES5 (because you're running it via the `gulp` CLI) but writing all your application files in ES6 (because you're compiling them through Babel).

**Comment by ilyavolodin** [comment]
Currently `es6` environment is a special case, you can't enable/disable it in inline comments. It's a known issue and we do not have a fix for it yet.
@gyandeeps You are correct, you can't do /* eslint-env es6:false */ that's just going to fail to parse, or will enable `es6:false` environment.

**Comment by gyandeeps** [comment]
It looks like a reasonable request/use case to me.

**Comment by ilyavolodin** [comment]
I think it's reasonable to be able to turn environments on and off based on inline config. es6 environment is a much larger problem though.

**Comment by nzakas** [comment]
Duplicate of #2134. This is only a problem for `es6`, all other environments already work

**Comment by IanVS** [comment]
@nzakas I'm not sure this is a duplicate.  As far as I can tell, #2134 was specifically regarding setting the `es6` environment inline.  This issue is about disabling environments, which I don't think work in any environment, including but not limited to es6.

For example, consider the file `test.js`:

```js
/*eslint no-undef: 2*/
/*eslint-env mocha:false*/

describe("Mocha is disabled");
```

```
eslint test.js --env mocha
```

Expected: error for undefined `describe`.

Actual Result: no errors

**Comment by ilyavolodin** [comment]
To the best of my recollection, you can only turn environments on. You can't turn them off.

**Comment by IanVS** [comment]
Yes, I believe this issue is requesting to add that enhancement.  It would be useful to me as well. (not volunteering right now though, sorry).

**Comment by IanVS** [comment]
@nzakas Can this issue be accepted?

**Comment by nzakas** [comment]
I'm on the fence on the utility of this. What do others think?

**Comment by IanVS** [comment]
I thought I needed it for the rule documentation linting, since all of those files need to stay in the same folder (`docs/rules/`), and some rules need ES6 while others are required to not have ES6 enabled.  But, now that I think about it, enabling es6 in the rules which do need it (which is now possible) is likely a better solution for this particular case.

There are a few issue comments out there suggesting to set `/*eslint-env es6:false*/`, however. (https://github.com/eslint/eslint/issues/2134#issuecomment-85230134 for example).

**Comment by callumlocke** [comment]
I can see why it's impossible to switch environments halfway through a file, because switching environments probably implies a different parser.

But there are obvious cases where I need to switch environments for a particular file, or for a particular directory, e.g. a single project containing both ES6 files for running in Node 4, and ES5 files for running in the browser... Is there any way I can do this today? It doesn't have to be a comment at the top of the file. What about having an extra .eslintrc file in the relevant directory, just to override the environment - would that work, or would I also need to repeat all the other config from the root .eslintrc?

**Comment by ilyavolodin** [comment]
@callumlocke Having another .eslintrc file in subdirectory will work. ESLint cascades all config files, so you can just turn off unneeded environments or add new ones for subdirectories.

**Comment by nzakas** [comment]
Closing in favor of #3698

**Comment by rocketmonkeys** [comment]
@ilyavolodin - Just confirming your last comment...
> Having another .eslintrc file in subdirectory will work. ESLint cascades all config files, so you can just turn off unneeded environments or add new ones for subdirectories.

I have a folder where I enable "node" environment at the root eslintrc, but then turn it off with a subfolder's esilntrc.  I want the subfolder to be client-files only (so "browser" env yes, "node" no).  From your comment it sounds like I can put "env: { node: false }" in the subfolder's eslintrc file and that will turn off the env, but it looks like that doesn't do it.  Is it really possible to turn off unneeded environments in the current code?

**Comment by rocketmonkeys** [comment]
@ilyavolodin - nevermind, I just saw #3915 - it is broken as I thought.  I'll post more on that ticket.

**Comment by ejosafat** [comment]
So in the end there is no plan to add this feature (disabling an env in a comment)? It's quite useful when you are gradually migrating your app from ES5 and ES6 and have a mix of both in the same folder, and it's always better to remove comments disabling es6 as you migrate rather than adding them to enable it.

**Comment by demisx** [comment]
Looks like this is now availabe, at least in v3.3.1 I am using:

```js
/* eslint-env es6 */
```

More here: http://eslint.org/docs/user-guide/configuring

**Comment by IanVS** [comment]
@demisx yes you can add environments inline, but not disable them, which is what this issue was about.

**Comment by demisx** [comment]
@IanVS Oh, I see. Sorry for misunderstanding.

**Comment by kachkaev** [comment]
Hey guys! Does a solution exist?

Calling [Nightmare's](https://github.com/segmentio/nightmare) `evaluate()` method inside a nodejs script to collect some info from the DOM. Looking for something like a 'local global' or a local `eslint-env`, sot that I could refer to `document` within `evaluate`.

Current solution relies on tweaking `no-undef`, but it's not ideal.

```js
nightmare
  .goto(url)
  .wait('#main')
  .evaluate(() => {
    /* eslint-disable no-undef */
    const mainElem = document.querySelector('#main');
    /* eslint-enable no-undef */
    // ...
  });
```
